File: Python/linkedlist/DeleteDuplicates.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution:
    def deleteDuplicates(self, head):
        """
        :type head: ListNode
        :rtype: ListNode
        """
        if not head or not head.next:
            return head
        pre, cur = head, head.next
        while cur:
            nextt = cur.next
            if pre.val == cur.val:
                pre.next = nextt
                cur = nextt
                
            else:
                pre = cur
                cur = nextt
                
        return head

File: JavaScript/linkedlist/DeleteDuplicates.js
/**
 * Definition for singly-linked list.
 * function ListNode(val) {
 *     this.val = val;
 *     this.next = null;
 * }
 */
/**
 * @param {ListNode} head
 * @return {ListNode}
 */
var deleteDuplicates = function(head) {
  let dummy = new ListNode(99999);
  dummy.next = head;
  let prev = dummy, curr = head;
  while (curr !== null) {
    if (prev.val === curr.val) {
      prev.next = curr.next;
    } else {
      prev = curr;
    }
    curr = curr.next;
  }
  return dummy.next;
};